client/src/components/Markdown/Elements/HistoryDatasetDisplay.vue

```diff
@@ -55,49 +55,46 @@
         </b-card-header>
         <b-card-body>
             <UrlDataProvider v-slot="{ result: itemContent, loading, error }" :url="itemUrl">
-                <UrlDataProvider v-slot="{ result: datatypesModel, loading: datatypesLoading }" :url="datatypesUrl">
-                    <LoadingSpan v-if="loading" message="Loading Dataset" />
-                    <LoadingSpan v-else-if="datatypesLoading" message="Loading Datatypes" />
-                    <div v-else-if="error">{{ error }}</div>
-                    <div v-else :class="contentClass">
-                        <b-embed
-                            v-if="isSubTypeOfAny(datasetType, ['pdf', 'html'], datatypesModel)"
-                            type="iframe"
-                            aspect="16by9"
-                            :src="displayUrl" />
-                        <HistoryDatasetAsImage
-                            v-else-if="isSubTypeOfAny(datasetType, ['galaxy.datatypes.images.Image'], datatypesModel)"
-                            :args="args" />
-                        <div v-else-if="itemContent.item_data">
-                            <div v-if="isSubTypeOfAny(datasetType, ['tabular'], datatypesModel)">
-                                <UrlDataProvider
-                                    v-slot="{ result: metaData, loading: metaLoading, error: metaError }"
-                                    :url="metaUrl">
-                                    <LoadingSpan v-if="metaLoading" message="Loading Metadata" />
-                                    <div v-else-if="metaError">{{ metaError }}</div>
-                                    <b-table
-                                        v-else
-                                        striped
-                                        hover
-                                        :fields="getFields(metaData)"
-                                        :items="getItems(itemContent.item_data, metaData)" />
-                                </UrlDataProvider>
-                            </div>
-                            <pre v-else>
-                                <code class="word-wrap-normal">{{ itemContent.item_data }}</code>
-                            </pre>
+                <LoadingSpan v-if="loading" message="Loading Dataset" />
+                <LoadingSpan v-else-if="datatypesLoading" message="Loading Datatypes" />
+                <div v-else-if="error">{{ error }}</div>
+                <div v-else :class="contentClass">
+                    <b-embed
+                        v-if="datatypesMapper?.isSubTypeOfAny(datasetType, ['pdf', 'html'])"
```

Comment:
The optional chaining here should've been a head-scratcher, it was hiding the real problem of not passing the datatypesMapper in the props.

Reply:
Just remove the optional chaining as in the diff, that'll cause plenty of errors to appear in the console. You're already making sure that the mapper is loaded on line 59.